src/NServiceBus.Core/Routing/RoutingPluginSettingsExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus
+{
+    using Features;
+    using Routing;
+
+    /// <summary>
+    /// Provides convinient API for plugins that need to extend the routing.
+    /// </summary>
+    public static class RoutingPluginSettingsExtensions
+    {
+        /// <summary>
+        /// Returns the routing table.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static UnicastRoutingTable RoutingTable(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<UnicastRoutingTable>();
+        }
+
+        /// <summary>
+        /// Returns the routing table.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static EndpointInstances EndpointInstances(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<EndpointInstances>();
+        }
+
+        /// <summary>
+        /// Returns the distribution policy.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static DistributionPolicy DistributionPolicy(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<DistributionPolicy>();
+        }
+    }
+}
```

Comment:
@timbussmann there is a sample that shows how to extend the routing (via a feature) and @mikeminutillo pointed out it would be nice to have such extensions rather than poking the settings with a magic wand.